### PR TITLE
Add docs for GitHub auth

### PR DESCRIPTION
**Pull request by thenickcox**
I can never remember the process of enabling OAuth for GitHub
correctly. It always takes a good bit of Googling and messing with
the URL and callback URL. Additionally, if I go to the trouble of forking and
cloning the repo, I want to get hacking right away. These docs
will streamline the process of getting the app set up for OAuth
with GitHub.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,8 +32,24 @@ $ bundle install
 
 ## Setup Authentication
 
-Need to set environment variables for login with github.
-Obtain the app_id and secret_token from https://github.com/settings/applications
+You'll need to set environment variables for login with GitHub. To do so,
+visit https://github.com/settings/applications.
+
+Set up your app with these values:
```
Comment (prathamesh-sonpatki) [inline README.md]: Can we rephrase it with `Create a new app with following values:`

Comment (thenickcox) [inline README.md]: But if you've cloned the repo, you're not really _creating_ the app; 'rails new' has already been run. You're adding to the already created app.

Comment (prathamesh-sonpatki) [inline README.md]: I meant the Github app. That's what the section explains right?

**Comment by thenickcox** [comment]
@prathamesh-sonpatki Good call. Added, squashed.

**Comment by prathamesh-sonpatki** [comment]
@thenickcox :heart: Thanx!

**Comment by schneems** [comment]
Thanks!

**Pull request merged**